src/components/dateTimePickerInput.js

```diff
@@ -253,6 +254,7 @@
         helperText={currentHelperText}
         disableToolbar={disableToolbar}
         disablePast={disablePastDates}
+        disableFuture={disableFutureDates}
```

Comment:
That could be a solution, but most of the time it's either the full future or the full past. I also wanted to follow the same style as the past. maxDate is also only if you'd want relative max date, if you'd want the complete future it would still be needed. This could also be applied for the past being too restricted as it is now.

How I see it, it would be an addition to the checkboxes.